owid/datautils/io/df.py

```diff
@@ -0,0 +1,84 @@
+"""DataFrame io operations."""
+import pandas as pd
+from pathlib import Path
+from owid.datautils.decorators import enable_file_download
+from typing import Any, Optional, Union, List
+
+
+COMPRESSION_SUPPORTED = ["gz", "bz2", "zip", "xz", "zst", "tar"]
+
+
+@enable_file_download("file_path")
+def from_file(
+    file_path: Union[str, Path], file_type: Optional[str] = None, **kwargs: Any
+) -> Union[pd.DataFrame, List[pd.DataFrame]]:
+    """Read a CSV or excel file.
+
+    It uses standard pandas function pandas.read_* but adds the ability to read from a URL in some
+    cases where pandas does not work.
+
+    The function will infer the extension of `file_path` by simply taking what follows the last ".". For example:
+    "file.csv" will have an extension ".csv"; "http://my/domain/file.xlsx" will have an extension "xlsx".
+
+    Reading from compressed files will not work by default, unless you provide a `file_type`.
```

Comment:
This was confusing me. It should be specified that the file type required is of the containing file (I thought it meant whether it was `zip` or `tar` etc.). And couldn't this be inferred, by looking at the content of the compressed object?

Reply:
Maybe it could attempt to infer the content, or assume it's a csv, and if it fails then a file_type should be specified. That would be more convenient than always imposing that file_type needs to be given.

Reply:
Let's force the user to specify the type for now. 

Feel free to create a new PR, that adds inferring support. Basically calling `decompres_file` and checking filetype.